src/main.rs

```diff
@@ -1,42 +1,137 @@
+        filename,
+        &source,
+        ParserOptions {
+            nesting: cli_args.nesting,
+            css_modules: cli_args.css_modules,
+            ..ParserOptions::default()
+        },
+    )
+    .unwrap();
+
+    if cli_args.minify {
+        stylesheet.minify(MinifyOptions::default()).unwrap();
     }
-  }
-}
 
-fn minify_css(cli_args: MinifyArgs) -> Result<(), std::io::Error> {
-  let source = fs::read_to_string(&cli_args.input_file)?;
-  let mut stylesheet = StyleSheet::parse(cli_args.input_file.to_string(), &source, ParserOptions::default()).unwrap();
-  stylesheet.minify(MinifyOptions::default()).unwrap();
-  let res = stylesheet.to_css(PrinterOptions { minify: true, ..PrinterOptions::default()}).unwrap();
+    let mut res = stylesheet
+        .to_css(PrinterOptions {
+            minify: cli_args.minify,
+            source_map: cli_args.sourcemap,
+            ..PrinterOptions::default()
+        })
+        .unwrap();
+
+    let map = if let Some(ref mut source_map) = res.source_map {
+        source_map.set_source_content(0, &res.code).map_err(|_| io::Error::new(io::ErrorKind::Other, "Error setting sourcemap"))?;
+        let mut vlq_output: Vec<u8> = Vec::new();
+        source_map.write_vlq(&mut vlq_output).map_err(|_| io::Error::new(io::ErrorKind::Other, "Error writing sourcemap vlq"))?;
+
+        let sm = SourceMapJson {
+            version: 3,
+            mappings: unsafe { String::from_utf8_unchecked(vlq_output) },
+            sources: source_map.get_sources(),
+            sources_content: source_map.get_sources_content(),
+            names: source_map.get_names()
+        };
+
+        serde_json::to_vec(&sm).ok()
+    } else {
+        None
+    };
+    if let Some(output_file) = &cli_args.output_file {
+        let mut code = res.code;
+        if cli_args.sourcemap {
+            if let Some(map_buf) = map {
+                let map_filename: String = output_file.to_owned() + ".map";
+                code += &format!("\n/*# sourceMappingURL={} */\n", map_filename);
+                fs::write(map_filename, map_buf)?;
+            }
+        }
+
+        fs::write(output_file, code.as_bytes())?;
 
-  if let Some(output_file) = cli_args.output_file {
-    fs::write(output_file, res.code.as_bytes())?;
-  } else {
-    println!("{}", res.code);
-  }
+        if cli_args.css_modules {
+            let css_modules_filename = cli_args
+                .css_modules_output_file
+                .unwrap_or(infer_css_modules_filename(cli_args.output_file)?);
+            if let Some(exports) = res.exports {
+                let css_modules_json = serde_json::to_string(&exports)?;
+                fs::write(css_modules_filename, css_modules_json)?;
+            }
+        }
 
-  Ok(())
+
+    } else {
+        println!("{}", res.code);
+        if cli_args.css_modules {
+            let css_modules_json = serde_json::to_string(&res.exports)?;
+            println!("{}", css_modules_json);
```

Comment:
printing css modules to stdout feels a little weird, but I guess if you add the `--css-modules` flag, you're expecting css modules, so better to get it than not?

Maybe better to require the `--output-file` flag in order to turn on `--css-modules`. Thoughts?

Reply:
Yeah I think requiring output file for css modules makes sense. Same with source maps, unless we inline the whole thing. I guess the main usecase for stdout is to pipe it somewhere, so probably most useful if that only includes the css.

Reply:
I'll make this switch! Drop the complexity / printing here, and make an output file required for using the css modules flag.